bgpd: Expand 'bgp default <afi>-<safi>' cmds

taspelund opened the pull request:
Historically the only address-family that could be enabled by default was IPv4 Unicast.
This was recently expanded in #8120 to include IPv6 Unicast.
This PR further expands this capability by including additional address-families that can be enabled by default.
e.g.
```
ub18(config-router)# bgp default 
  ipv4-flowspec           Activate ipv4-flowspec for a peer by default
  ipv4-multicast          Activate ipv4-multicast for a peer by default
  ipv4-unicast            Activate ipv4-unicast for a peer by default
  ipv4-vpn                Activate ipv4-vpn for a peer by default
  ipv6-flowspec           Activate ipv6-flowspec for a peer by default
  ipv6-multicast          Activate ipv6-multicast for a peer by default
  ipv6-unicast            Activate ipv6-unicast for a peer by default
  ipv6-vpn                Activate ipv6-vpn for a peer by default
  l2vpn-evpn              Activate l2vpn-evpn for a peer by default
```

polychaeta reviewed (commented):
Thanks for your contribution to FRR!


<details>
<summary><b>Click for style suggestions</b></summary>

To apply these suggestions:

<p>

```
curl -s https://gist.githubusercontent.com/polychaeta/883cceb9572fee1661226e2bb422dd5f/raw/148e78bd22d34126b0c1b0d087c2e5e16797b496/cr_8401_1617594558.diff | git apply
```

</p>

<p>

```diff
diff --git a/bgpd/bgpd.h b/bgpd/bgpd.h
index a92664cdd..32f42c544 100644
--- a/bgpd/bgpd.h
+++ b/bgpd/bgpd.h
@@ -449,50 +449,50 @@ struct bgp {
 #define BGP_FLAG_DETERMINISTIC_MED        (1 << 1)
 #define BGP_FLAG_MED_MISSING_AS_WORST     (1 << 2)
 #define BGP_FLAG_MED_CONFED               (1 << 3)
-#define BGP_FLAG_NO_CLIENT_TO_CLIENT      (1 << 4)
-#define BGP_FLAG_COMPARE_ROUTER_ID        (1 << 5)
-#define BGP_FLAG_ASPATH_IGNORE            (1 << 6)
-#define BGP_FLAG_IMPORT_CHECK             (1 << 7)
-#define BGP_FLAG_NO_FAST_EXT_FAILOVER     (1 << 8)
-#define BGP_FLAG_LOG_NEIGHBOR_CHANGES     (1 << 9)
+#define BGP_FLAG_NO_CLIENT_TO_CLIENT (1 << 4)
+#define BGP_FLAG_COMPARE_ROUTER_ID (1 << 5)
+#define BGP_FLAG_ASPATH_IGNORE (1 << 6)
+#define BGP_FLAG_IMPORT_CHECK (1 << 7)
+#define BGP_FLAG_NO_FAST_EXT_FAILOVER (1 << 8)
+#define BGP_FLAG_LOG_NEIGHBOR_CHANGES (1 << 9)
 
 /* This flag is set when we have full BGP Graceful-Restart mode enable */
-#define BGP_FLAG_GRACEFUL_RESTART         (1 << 10)
+#define BGP_FLAG_GRACEFUL_RESTART (1 << 10)
 
-#define BGP_FLAG_ASPATH_CONFED            (1 << 11)
-#define BGP_FLAG_ASPATH_MULTIPATH_RELAX   (1 << 12)
+#define BGP_FLAG_ASPATH_CONFED (1 << 11)
+#define BGP_FLAG_ASPATH_MULTIPATH_RELAX (1 << 12)
 #define BGP_FLAG_RR_ALLOW_OUTBOUND_POLICY (1 << 13)
 #define BGP_FLAG_DISABLE_NH_CONNECTED_CHK (1 << 14)
-#define BGP_FLAG_MULTIPATH_RELAX_AS_SET   (1 << 15)
-#define BGP_FLAG_FORCE_STATIC_PROCESS     (1 << 16)
-#define BGP_FLAG_SHOW_HOSTNAME            (1 << 17)
-#define BGP_FLAG_GR_PRESERVE_FWD          (1 << 18)
-#define BGP_FLAG_GRACEFUL_SHUTDOWN        (1 << 19)
-#define BGP_FLAG_DELETE_IN_PROGRESS       (1 << 20)
-#define BGP_FLAG_SELECT_DEFER_DISABLE     (1 << 21)
-#define BGP_FLAG_GR_DISABLE_EOR           (1 << 22)
-#define BGP_FLAG_EBGP_REQUIRES_POLICY     (1 << 23)
-#define BGP_FLAG_SHOW_NEXTHOP_HOSTNAME    (1 << 24)
+#define BGP_FLAG_MULTIPATH_RELAX_AS_SET (1 << 15)
+#define BGP_FLAG_FORCE_STATIC_PROCESS (1 << 16)
+#define BGP_FLAG_SHOW_HOSTNAME (1 << 17)
+#define BGP_FLAG_GR_PRESERVE_FWD (1 << 18)
+#define BGP_FLAG_GRACEFUL_SHUTDOWN (1 << 19)
+#define BGP_FLAG_DELETE_IN_PROGRESS (1 << 20)
+#define BGP_FLAG_SELECT_DEFER_DISABLE (1 << 21)
+#define BGP_FLAG_GR_DISABLE_EOR (1 << 22)
+#define BGP_FLAG_EBGP_REQUIRES_POLICY (1 << 23)
+#define BGP_FLAG_SHOW_NEXTHOP_HOSTNAME (1 << 24)
 
 /* This flag is set if the instance is in administrative shutdown */
-#define BGP_FLAG_SHUTDOWN                 (1 << 25)
-#define BGP_FLAG_SUPPRESS_FIB_PENDING     (1 << 26)
-#define BGP_FLAG_SUPPRESS_DUPLICATES      (1 << 27)
+#define BGP_FLAG_SHUTDOWN (1 << 25)
+#define BGP_FLAG_SUPPRESS_FIB_PENDING (1 << 26)
+#define BGP_FLAG_SUPPRESS_DUPLICATES (1 << 27)
 #define BGP_FLAG_PEERTYPE_MULTIPATH_RELAX (1 << 29)
 
-       /* BGP default address-families.
-        * New peers inherit enabled afi/safis from bgp instance.
-        */
-       uint16_t default_af_flags;
-#define BGP_DEFAULT_AF_IPV4_UNICAST            (1 << 0)
-#define BGP_DEFAULT_AF_IPV4_MULTICAST          (1 << 1)
-#define BGP_DEFAULT_AF_IPV4_MPLS_VPN           (1 << 2)
-#define BGP_DEFAULT_AF_IPV4_FLOWSPEC           (1 << 3)
-#define BGP_DEFAULT_AF_IPV6_UNICAST            (1 << 4)
-#define BGP_DEFAULT_AF_IPV6_MULTICAST          (1 << 5)
-#define BGP_DEFAULT_AF_IPV6_MPLS_VPN           (1 << 6)
-#define BGP_DEFAULT_AF_IPV6_FLOWSPEC           (1 << 7)
-#define BGP_DEFAULT_AF_L2VPN_EVPN              (1 << 8)
+	/* BGP default address-families.
+	 * New peers inherit enabled afi/safis from bgp instance.
+	 */
+	uint16_t default_af_flags;
+#define BGP_DEFAULT_AF_IPV4_UNICAST (1 << 0)
+#define BGP_DEFAULT_AF_IPV4_MULTICAST (1 << 1)
+#define BGP_DEFAULT_AF_IPV4_MPLS_VPN (1 << 2)
+#define BGP_DEFAULT_AF_IPV4_FLOWSPEC (1 << 3)
+#define BGP_DEFAULT_AF_IPV6_UNICAST (1 << 4)
+#define BGP_DEFAULT_AF_IPV6_MULTICAST (1 << 5)
+#define BGP_DEFAULT_AF_IPV6_MPLS_VPN (1 << 6)
+#define BGP_DEFAULT_AF_IPV6_FLOWSPEC (1 << 7)
+#define BGP_DEFAULT_AF_L2VPN_EVPN (1 << 8)
 
 	enum global_mode GLOBAL_GR_FSM[BGP_GLOBAL_GR_MODE]
 				      [BGP_GLOBAL_GR_EVENT_CMD];

```

</p>
</details>

            
If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

mwinter-osr commented:
CI:rerun

polychaeta reviewed (commented):
Thanks for your contribution to FRR!


<details>
<summary><b>Click for style suggestions</b></summary>

To apply these suggestions:

<p>

```
curl -s https://gist.githubusercontent.com/polychaeta/a60437c2b01e27134be4d89d233fcf6e/raw/5727a491eda3fb82e55c8f688e03690139edcdff/cr_8401_1617625938.diff | git apply
```

</p>

<p>

```diff
diff --git a/bgpd/bgp_vty.c b/bgpd/bgp_vty.c
index 1046549fb..7a2a7562e 100644
--- a/bgpd/bgp_vty.c
+++ b/bgpd/bgp_vty.c
@@ -17148,58 +17148,50 @@ static void bgp_config_write_peer_af(struct vty *vty, struct bgp *bgp,
 		if (peer->afc[afi][safi]) {
 			if (afi == AFI_IP) {
 				if ((safi == SAFI_UNICAST)
-				    && !CHECK_FLAG(
-					    bgp->default_af_flags,
-					    BGP_DEFAULT_AF_IPV4_UNICAST))
+				    && !CHECK_FLAG(bgp->default_af_flags,
+						   BGP_DEFAULT_AF_IPV4_UNICAST))
 					vty_out(vty, "  neighbor %s activate\n",
 						addr);
-				else if (
-					(safi == SAFI_MULTICAST)
-					&& !CHECK_FLAG(
-						bgp->default_af_flags,
-						BGP_DEFAULT_AF_IPV4_MULTICAST))
+				else if ((safi == SAFI_MULTICAST)
+					 && !CHECK_FLAG(
+						 bgp->default_af_flags,
+						 BGP_DEFAULT_AF_IPV4_MULTICAST))
 					vty_out(vty, "  neighbor %s activate\n",
 						addr);
-				else if (
-					(safi == SAFI_MPLS_VPN)
-					&& !CHECK_FLAG(
-						bgp->default_af_flags,
-						BGP_DEFAULT_AF_IPV4_MPLS_VPN))
+				else if ((safi == SAFI_MPLS_VPN)
+					 && !CHECK_FLAG(
+						 bgp->default_af_flags,
+						 BGP_DEFAULT_AF_IPV4_MPLS_VPN))
 					vty_out(vty, "  neighbor %s activate\n",
 						addr);
-				else if (
-					(safi == SAFI_FLOWSPEC)
-					&& !CHECK_FLAG(
-						bgp->default_af_flags,
-						BGP_DEFAULT_AF_IPV4_FLOWSPEC))
+				else if ((safi == SAFI_FLOWSPEC)
+					 && !CHECK_FLAG(
+						 bgp->default_af_flags,
+						 BGP_DEFAULT_AF_IPV4_FLOWSPEC))
 					vty_out(vty, "  neighbor %s activate\n",
 						addr);
 			} else if (afi == AFI_IP6) {
 				if ((safi == SAFI_UNICAST)
-				    && !CHECK_FLAG(
-					    bgp->default_af_flags,
-					    BGP_DEFAULT_AF_IPV6_UNICAST))
+				    && !CHECK_FLAG(bgp->default_af_flags,
+						   BGP_DEFAULT_AF_IPV6_UNICAST))
 					vty_out(vty, "  neighbor %s activate\n",
 						addr);
-				else if (
-					(safi == SAFI_MULTICAST)
-					&& !CHECK_FLAG(
-						bgp->default_af_flags,
-						BGP_DEFAULT_AF_IPV6_MULTICAST))
+				else if ((safi == SAFI_MULTICAST)
+					 && !CHECK_FLAG(
+						 bgp->default_af_flags,
+						 BGP_DEFAULT_AF_IPV6_MULTICAST))
 					vty_out(vty, "  neighbor %s activate\n",
 						addr);
-				else if (
-					(safi == SAFI_MPLS_VPN)
-					&& !CHECK_FLAG(
-						bgp->default_af_flags,
-						BGP_DEFAULT_AF_IPV6_MPLS_VPN))
+				else if ((safi == SAFI_MPLS_VPN)
+					 && !CHECK_FLAG(
+						 bgp->default_af_flags,
+						 BGP_DEFAULT_AF_IPV6_MPLS_VPN))
 					vty_out(vty, "  neighbor %s activate\n",
 						addr);
-				else if (
-					(safi == SAFI_FLOWSPEC)
-					&& !CHECK_FLAG(
-						bgp->default_af_flags,
-						BGP_DEFAULT_AF_IPV6_FLOWSPEC))
+				else if ((safi == SAFI_FLOWSPEC)
+					 && !CHECK_FLAG(
+						 bgp->default_af_flags,
+						 BGP_DEFAULT_AF_IPV6_FLOWSPEC))
 					vty_out(vty, "  neighbor %s activate\n",
 						addr);
 			} else if (afi == AFI_L2VPN && safi == SAFI_EVPN) {
@@ -17213,65 +17205,57 @@ static void bgp_config_write_peer_af(struct vty *vty, struct bgp *bgp,
 		} else {
 			if (afi == AFI_IP) {
 				if ((safi == SAFI_UNICAST)
-				    && CHECK_FLAG(
-					    bgp->default_af_flags,
-					    BGP_DEFAULT_AF_IPV4_UNICAST))
+				    && CHECK_FLAG(bgp->default_af_flags,
+						  BGP_DEFAULT_AF_IPV4_UNICAST))
 					vty_out(vty,
 						"  no neighbor %s activate\n",
 						addr);
-				else if (
-					(safi == SAFI_MULTICAST)
-					&& CHECK_FLAG(
-						bgp->default_af_flags,
-						BGP_DEFAULT_AF_IPV4_MULTICAST))
+				else if ((safi == SAFI_MULTICAST)
+					 && CHECK_FLAG(
+						 bgp->default_af_flags,
+						 BGP_DEFAULT_AF_IPV4_MULTICAST))
 					vty_out(vty,
 						"  no neighbor %s activate\n",
 						addr);
-				else if (
-					(safi == SAFI_MPLS_VPN)
-					&& CHECK_FLAG(
-						bgp->default_af_flags,
-						BGP_DEFAULT_AF_IPV4_MPLS_VPN))
+				else if ((safi == SAFI_MPLS_VPN)
+					 && CHECK_FLAG(
+						 bgp->default_af_flags,
+						 BGP_DEFAULT_AF_IPV4_MPLS_VPN))
 					vty_out(vty,
 						"  no neighbor %s activate\n",
 						addr);
-				else if (
-					(safi == SAFI_FLOWSPEC)
-					&& CHECK_FLAG(
-						bgp->default_af_flags,
-						BGP_DEFAULT_AF_IPV4_FLOWSPEC))
+				else if ((safi == SAFI_FLOWSPEC)
+					 && CHECK_FLAG(
+						 bgp->default_af_flags,
+						 BGP_DEFAULT_AF_IPV4_FLOWSPEC))
 					vty_out(vty,
 						"  no neighbor %s activate\n",
 						addr);
 			} else if (afi == AFI_IP6) {
 				if ((safi == SAFI_UNICAST)
-				    && CHECK_FLAG(
-					    bgp->default_af_flags,
-					    BGP_DEFAULT_AF_IPV6_UNICAST))
+				    && CHECK_FLAG(bgp->default_af_flags,
+						  BGP_DEFAULT_AF_IPV6_UNICAST))
 					vty_out(vty,
 						"  no neighbor %s activate\n",
 						addr);
-				else if (
-					(safi == SAFI_MULTICAST)
-					&& CHECK_FLAG(
-						bgp->default_af_flags,
-						BGP_DEFAULT_AF_IPV6_MULTICAST))
+				else if ((safi == SAFI_MULTICAST)
+					 && CHECK_FLAG(
+						 bgp->default_af_flags,
+						 BGP_DEFAULT_AF_IPV6_MULTICAST))
 					vty_out(vty,
 						"  no neighbor %s activate\n",
 						addr);
-				else if (
-					(safi == SAFI_MPLS_VPN)
-					&& CHECK_FLAG(
-						bgp->default_af_flags,
-						BGP_DEFAULT_AF_IPV6_MPLS_VPN))
+				else if ((safi == SAFI_MPLS_VPN)
+					 && CHECK_FLAG(
+						 bgp->default_af_flags,
+						 BGP_DEFAULT_AF_IPV6_MPLS_VPN))
 					vty_out(vty,
 						"  no neighbor %s activate\n",
 						addr);
-				else if (
-					(safi == SAFI_FLOWSPEC)
-					&& CHECK_FLAG(
-						bgp->default_af_flags,
-						BGP_DEFAULT_AF_IPV6_FLOWSPEC))
+				else if ((safi == SAFI_FLOWSPEC)
+					 && CHECK_FLAG(
+						 bgp->default_af_flags,
+						 BGP_DEFAULT_AF_IPV6_FLOWSPEC))
 					vty_out(vty,
 						"  no neighbor %s activate\n",
 						addr);
diff --git a/bgpd/bgpd.h b/bgpd/bgpd.h
index 1a64d60f0..32f42c544 100644
--- a/bgpd/bgpd.h
+++ b/bgpd/bgpd.h
@@ -449,50 +449,50 @@ struct bgp {
 #define BGP_FLAG_DETERMINISTIC_MED        (1 << 1)
 #define BGP_FLAG_MED_MISSING_AS_WORST     (1 << 2)
 #define BGP_FLAG_MED_CONFED               (1 << 3)
-#define BGP_FLAG_NO_CLIENT_TO_CLIENT      (1 << 4)
-#define BGP_FLAG_COMPARE_ROUTER_ID        (1 << 5)
-#define BGP_FLAG_ASPATH_IGNORE            (1 << 6)
-#define BGP_FLAG_IMPORT_CHECK             (1 << 7)
-#define BGP_FLAG_NO_FAST_EXT_FAILOVER     (1 << 8)
-#define BGP_FLAG_LOG_NEIGHBOR_CHANGES     (1 << 9)
+#define BGP_FLAG_NO_CLIENT_TO_CLIENT (1 << 4)
+#define BGP_FLAG_COMPARE_ROUTER_ID (1 << 5)
+#define BGP_FLAG_ASPATH_IGNORE (1 << 6)
+#define BGP_FLAG_IMPORT_CHECK (1 << 7)
+#define BGP_FLAG_NO_FAST_EXT_FAILOVER (1 << 8)
+#define BGP_FLAG_LOG_NEIGHBOR_CHANGES (1 << 9)
 
 /* This flag is set when we have full BGP Graceful-Restart mode enable */
-#define BGP_FLAG_GRACEFUL_RESTART         (1 << 10)
+#define BGP_FLAG_GRACEFUL_RESTART (1 << 10)
 
-#define BGP_FLAG_ASPATH_CONFED            (1 << 11)
-#define BGP_FLAG_ASPATH_MULTIPATH_RELAX   (1 << 12)
+#define BGP_FLAG_ASPATH_CONFED (1 << 11)
+#define BGP_FLAG_ASPATH_MULTIPATH_RELAX (1 << 12)
 #define BGP_FLAG_RR_ALLOW_OUTBOUND_POLICY (1 << 13)
 #define BGP_FLAG_DISABLE_NH_CONNECTED_CHK (1 << 14)
-#define BGP_FLAG_MULTIPATH_RELAX_AS_SET   (1 << 15)
-#define BGP_FLAG_FORCE_STATIC_PROCESS     (1 << 16)
-#define BGP_FLAG_SHOW_HOSTNAME            (1 << 17)
-#define BGP_FLAG_GR_PRESERVE_FWD          (1 << 18)
-#define BGP_FLAG_GRACEFUL_SHUTDOWN        (1 << 19)
-#define BGP_FLAG_DELETE_IN_PROGRESS       (1 << 20)
-#define BGP_FLAG_SELECT_DEFER_DISABLE     (1 << 21)
-#define BGP_FLAG_GR_DISABLE_EOR           (1 << 22)
-#define BGP_FLAG_EBGP_REQUIRES_POLICY     (1 << 23)
-#define BGP_FLAG_SHOW_NEXTHOP_HOSTNAME    (1 << 24)
+#define BGP_FLAG_MULTIPATH_RELAX_AS_SET (1 << 15)
+#define BGP_FLAG_FORCE_STATIC_PROCESS (1 << 16)
+#define BGP_FLAG_SHOW_HOSTNAME (1 << 17)
+#define BGP_FLAG_GR_PRESERVE_FWD (1 << 18)
+#define BGP_FLAG_GRACEFUL_SHUTDOWN (1 << 19)
+#define BGP_FLAG_DELETE_IN_PROGRESS (1 << 20)
+#define BGP_FLAG_SELECT_DEFER_DISABLE (1 << 21)
+#define BGP_FLAG_GR_DISABLE_EOR (1 << 22)
+#define BGP_FLAG_EBGP_REQUIRES_POLICY (1 << 23)
+#define BGP_FLAG_SHOW_NEXTHOP_HOSTNAME (1 << 24)
 
 /* This flag is set if the instance is in administrative shutdown */
-#define BGP_FLAG_SHUTDOWN                 (1 << 25)
-#define BGP_FLAG_SUPPRESS_FIB_PENDING     (1 << 26)
-#define BGP_FLAG_SUPPRESS_DUPLICATES      (1 << 27)
+#define BGP_FLAG_SHUTDOWN (1 << 25)
+#define BGP_FLAG_SUPPRESS_FIB_PENDING (1 << 26)
+#define BGP_FLAG_SUPPRESS_DUPLICATES (1 << 27)
 #define BGP_FLAG_PEERTYPE_MULTIPATH_RELAX (1 << 29)
 
 	/* BGP default address-families.
 	 * New peers inherit enabled afi/safis from bgp instance.
 	 */
 	uint16_t default_af_flags;
-#define BGP_DEFAULT_AF_IPV4_UNICAST            (1 << 0)
-#define BGP_DEFAULT_AF_IPV4_MULTICAST          (1 << 1)
-#define BGP_DEFAULT_AF_IPV4_MPLS_VPN           (1 << 2)
-#define BGP_DEFAULT_AF_IPV4_FLOWSPEC           (1 << 3)
-#define BGP_DEFAULT_AF_IPV6_UNICAST            (1 << 4)
-#define BGP_DEFAULT_AF_IPV6_MULTICAST          (1 << 5)
-#define BGP_DEFAULT_AF_IPV6_MPLS_VPN           (1 << 6)
-#define BGP_DEFAULT_AF_IPV6_FLOWSPEC           (1 << 7)
-#define BGP_DEFAULT_AF_L2VPN_EVPN              (1 << 8)
+#define BGP_DEFAULT_AF_IPV4_UNICAST (1 << 0)
+#define BGP_DEFAULT_AF_IPV4_MULTICAST (1 << 1)
+#define BGP_DEFAULT_AF_IPV4_MPLS_VPN (1 << 2)
+#define BGP_DEFAULT_AF_IPV4_FLOWSPEC (1 << 3)
+#define BGP_DEFAULT_AF_IPV6_UNICAST (1 << 4)
+#define BGP_DEFAULT_AF_IPV6_MULTICAST (1 << 5)
+#define BGP_DEFAULT_AF_IPV6_MPLS_VPN (1 << 6)
+#define BGP_DEFAULT_AF_IPV6_FLOWSPEC (1 << 7)
+#define BGP_DEFAULT_AF_L2VPN_EVPN (1 << 8)
 
 	enum global_mode GLOBAL_GR_FSM[BGP_GLOBAL_GR_MODE]
 				      [BGP_GLOBAL_GR_EVENT_CMD];

```

</p>
</details>

            
If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

polychaeta reviewed (commented):
Thanks for your contribution to FRR!


<details>
<summary><b>Click for style suggestions</b></summary>

To apply these suggestions:

<p>

```
curl -s https://gist.githubusercontent.com/polychaeta/a67ef59c95d6237b6a1c9061707a2b87/raw/da81a1ef4d42984f7b1fac599a0e70da587b94eb/cr_8401_1617626081.diff | git apply
```

</p>

<p>

```diff
diff --git a/bgpd/bgpd.h b/bgpd/bgpd.h
index 1a64d60f0..32f42c544 100644
--- a/bgpd/bgpd.h
+++ b/bgpd/bgpd.h
@@ -449,50 +449,50 @@ struct bgp {
 #define BGP_FLAG_DETERMINISTIC_MED        (1 << 1)
 #define BGP_FLAG_MED_MISSING_AS_WORST     (1 << 2)
 #define BGP_FLAG_MED_CONFED               (1 << 3)
-#define BGP_FLAG_NO_CLIENT_TO_CLIENT      (1 << 4)
-#define BGP_FLAG_COMPARE_ROUTER_ID        (1 << 5)
-#define BGP_FLAG_ASPATH_IGNORE            (1 << 6)
-#define BGP_FLAG_IMPORT_CHECK             (1 << 7)
-#define BGP_FLAG_NO_FAST_EXT_FAILOVER     (1 << 8)
-#define BGP_FLAG_LOG_NEIGHBOR_CHANGES     (1 << 9)
+#define BGP_FLAG_NO_CLIENT_TO_CLIENT (1 << 4)
+#define BGP_FLAG_COMPARE_ROUTER_ID (1 << 5)
+#define BGP_FLAG_ASPATH_IGNORE (1 << 6)
+#define BGP_FLAG_IMPORT_CHECK (1 << 7)
+#define BGP_FLAG_NO_FAST_EXT_FAILOVER (1 << 8)
+#define BGP_FLAG_LOG_NEIGHBOR_CHANGES (1 << 9)
 
 /* This flag is set when we have full BGP Graceful-Restart mode enable */
-#define BGP_FLAG_GRACEFUL_RESTART         (1 << 10)
+#define BGP_FLAG_GRACEFUL_RESTART (1 << 10)
 
-#define BGP_FLAG_ASPATH_CONFED            (1 << 11)
-#define BGP_FLAG_ASPATH_MULTIPATH_RELAX   (1 << 12)
+#define BGP_FLAG_ASPATH_CONFED (1 << 11)
+#define BGP_FLAG_ASPATH_MULTIPATH_RELAX (1 << 12)
 #define BGP_FLAG_RR_ALLOW_OUTBOUND_POLICY (1 << 13)
 #define BGP_FLAG_DISABLE_NH_CONNECTED_CHK (1 << 14)
-#define BGP_FLAG_MULTIPATH_RELAX_AS_SET   (1 << 15)
-#define BGP_FLAG_FORCE_STATIC_PROCESS     (1 << 16)
-#define BGP_FLAG_SHOW_HOSTNAME            (1 << 17)
-#define BGP_FLAG_GR_PRESERVE_FWD          (1 << 18)
-#define BGP_FLAG_GRACEFUL_SHUTDOWN        (1 << 19)
-#define BGP_FLAG_DELETE_IN_PROGRESS       (1 << 20)
-#define BGP_FLAG_SELECT_DEFER_DISABLE     (1 << 21)
-#define BGP_FLAG_GR_DISABLE_EOR           (1 << 22)
-#define BGP_FLAG_EBGP_REQUIRES_POLICY     (1 << 23)
-#define BGP_FLAG_SHOW_NEXTHOP_HOSTNAME    (1 << 24)
+#define BGP_FLAG_MULTIPATH_RELAX_AS_SET (1 << 15)
+#define BGP_FLAG_FORCE_STATIC_PROCESS (1 << 16)
+#define BGP_FLAG_SHOW_HOSTNAME (1 << 17)
+#define BGP_FLAG_GR_PRESERVE_FWD (1 << 18)
+#define BGP_FLAG_GRACEFUL_SHUTDOWN (1 << 19)
+#define BGP_FLAG_DELETE_IN_PROGRESS (1 << 20)
+#define BGP_FLAG_SELECT_DEFER_DISABLE (1 << 21)
+#define BGP_FLAG_GR_DISABLE_EOR (1 << 22)
+#define BGP_FLAG_EBGP_REQUIRES_POLICY (1 << 23)
+#define BGP_FLAG_SHOW_NEXTHOP_HOSTNAME (1 << 24)
 
 /* This flag is set if the instance is in administrative shutdown */
-#define BGP_FLAG_SHUTDOWN                 (1 << 25)
-#define BGP_FLAG_SUPPRESS_FIB_PENDING     (1 << 26)
-#define BGP_FLAG_SUPPRESS_DUPLICATES      (1 << 27)
+#define BGP_FLAG_SHUTDOWN (1 << 25)
+#define BGP_FLAG_SUPPRESS_FIB_PENDING (1 << 26)
+#define BGP_FLAG_SUPPRESS_DUPLICATES (1 << 27)
 #define BGP_FLAG_PEERTYPE_MULTIPATH_RELAX (1 << 29)
 
 	/* BGP default address-families.
 	 * New peers inherit enabled afi/safis from bgp instance.
 	 */
 	uint16_t default_af_flags;
-#define BGP_DEFAULT_AF_IPV4_UNICAST            (1 << 0)
-#define BGP_DEFAULT_AF_IPV4_MULTICAST          (1 << 1)
-#define BGP_DEFAULT_AF_IPV4_MPLS_VPN           (1 << 2)
-#define BGP_DEFAULT_AF_IPV4_FLOWSPEC           (1 << 3)
-#define BGP_DEFAULT_AF_IPV6_UNICAST            (1 << 4)
-#define BGP_DEFAULT_AF_IPV6_MULTICAST          (1 << 5)
-#define BGP_DEFAULT_AF_IPV6_MPLS_VPN           (1 << 6)
-#define BGP_DEFAULT_AF_IPV6_FLOWSPEC           (1 << 7)
-#define BGP_DEFAULT_AF_L2VPN_EVPN              (1 << 8)
+#define BGP_DEFAULT_AF_IPV4_UNICAST (1 << 0)
+#define BGP_DEFAULT_AF_IPV4_MULTICAST (1 << 1)
+#define BGP_DEFAULT_AF_IPV4_MPLS_VPN (1 << 2)
+#define BGP_DEFAULT_AF_IPV4_FLOWSPEC (1 << 3)
+#define BGP_DEFAULT_AF_IPV6_UNICAST (1 << 4)
+#define BGP_DEFAULT_AF_IPV6_MULTICAST (1 << 5)
+#define BGP_DEFAULT_AF_IPV6_MPLS_VPN (1 << 6)
+#define BGP_DEFAULT_AF_IPV6_FLOWSPEC (1 << 7)
+#define BGP_DEFAULT_AF_L2VPN_EVPN (1 << 8)
 
 	enum global_mode GLOBAL_GR_FSM[BGP_GLOBAL_GR_MODE]
 				      [BGP_GLOBAL_GR_EVENT_CMD];

```

</p>
</details>

            
If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

ton31337 commented:
Can you extend https://github.com/FRRouting/frr/pull/8120/commits/484ff4899b1d1083471dd7fe100044ea7a755056 this test with at least one of the families you appended?

rwestphal reviewed (commented):
I like the idea but I think we could avoid lots of code duplication by using `FOREACH_AFI_SAFI` and a wrapper function that maps AFIs/SAFIs to `BGP_DEFAULT_AF_*` constants. Maybe we could have a single "bgp default" command as well.

taspelund commented:
Thanks for the feedback guys.  I'll look into both items and update the PR.

polychaeta reviewed (changes requested):
Thanks for your contribution to FRR!

* `sprintf, strcat, strcpy, inet_ntoa` are banned; please use `snprintf, strlcat, strlcpy, inet_ntop`

<details>
<summary><b>Click for style suggestions</b></summary>

To apply these suggestions:

<p>

```
curl -s https://gist.githubusercontent.com/polychaeta/6681ec1706b04f53a08293337739d2cb/raw/19a1e86a4aa0c1ba66271f570f0d42aa59fe3480/cr_8401_1620050529.diff | git apply
```

</p>

<p>

```diff
diff --git a/bgpd/bgpd.h b/bgpd/bgpd.h
index 1e6624fc9..14e8141e1 100644
--- a/bgpd/bgpd.h
+++ b/bgpd/bgpd.h
@@ -450,50 +450,50 @@ struct bgp {
 #define BGP_FLAG_DETERMINISTIC_MED        (1 << 1)
 #define BGP_FLAG_MED_MISSING_AS_WORST     (1 << 2)
 #define BGP_FLAG_MED_CONFED               (1 << 3)
-#define BGP_FLAG_NO_CLIENT_TO_CLIENT      (1 << 4)
-#define BGP_FLAG_COMPARE_ROUTER_ID        (1 << 5)
-#define BGP_FLAG_ASPATH_IGNORE            (1 << 6)
-#define BGP_FLAG_IMPORT_CHECK             (1 << 7)
-#define BGP_FLAG_NO_FAST_EXT_FAILOVER     (1 << 8)
-#define BGP_FLAG_LOG_NEIGHBOR_CHANGES     (1 << 9)
+#define BGP_FLAG_NO_CLIENT_TO_CLIENT (1 << 4)
+#define BGP_FLAG_COMPARE_ROUTER_ID (1 << 5)
+#define BGP_FLAG_ASPATH_IGNORE (1 << 6)
+#define BGP_FLAG_IMPORT_CHECK (1 << 7)
+#define BGP_FLAG_NO_FAST_EXT_FAILOVER (1 << 8)
+#define BGP_FLAG_LOG_NEIGHBOR_CHANGES (1 << 9)
 
 /* This flag is set when we have full BGP Graceful-Restart mode enable */
-#define BGP_FLAG_GRACEFUL_RESTART         (1 << 10)
+#define BGP_FLAG_GRACEFUL_RESTART (1 << 10)
 
-#define BGP_FLAG_ASPATH_CONFED            (1 << 11)
-#define BGP_FLAG_ASPATH_MULTIPATH_RELAX   (1 << 12)
+#define BGP_FLAG_ASPATH_CONFED (1 << 11)
+#define BGP_FLAG_ASPATH_MULTIPATH_RELAX (1 << 12)
 #define BGP_FLAG_RR_ALLOW_OUTBOUND_POLICY (1 << 13)
 #define BGP_FLAG_DISABLE_NH_CONNECTED_CHK (1 << 14)
-#define BGP_FLAG_MULTIPATH_RELAX_AS_SET   (1 << 15)
-#define BGP_FLAG_FORCE_STATIC_PROCESS     (1 << 16)
-#define BGP_FLAG_SHOW_HOSTNAME            (1 << 17)
-#define BGP_FLAG_GR_PRESERVE_FWD          (1 << 18)
-#define BGP_FLAG_GRACEFUL_SHUTDOWN        (1 << 19)
-#define BGP_FLAG_DELETE_IN_PROGRESS       (1 << 20)
-#define BGP_FLAG_SELECT_DEFER_DISABLE     (1 << 21)
-#define BGP_FLAG_GR_DISABLE_EOR           (1 << 22)
-#define BGP_FLAG_EBGP_REQUIRES_POLICY     (1 << 23)
-#define BGP_FLAG_SHOW_NEXTHOP_HOSTNAME    (1 << 24)
+#define BGP_FLAG_MULTIPATH_RELAX_AS_SET (1 << 15)
+#define BGP_FLAG_FORCE_STATIC_PROCESS (1 << 16)
+#define BGP_FLAG_SHOW_HOSTNAME (1 << 17)
+#define BGP_FLAG_GR_PRESERVE_FWD (1 << 18)
+#define BGP_FLAG_GRACEFUL_SHUTDOWN (1 << 19)
+#define BGP_FLAG_DELETE_IN_PROGRESS (1 << 20)
+#define BGP_FLAG_SELECT_DEFER_DISABLE (1 << 21)
+#define BGP_FLAG_GR_DISABLE_EOR (1 << 22)
+#define BGP_FLAG_EBGP_REQUIRES_POLICY (1 << 23)
+#define BGP_FLAG_SHOW_NEXTHOP_HOSTNAME (1 << 24)
 
 /* This flag is set if the instance is in administrative shutdown */
-#define BGP_FLAG_SHUTDOWN                 (1 << 25)
-#define BGP_FLAG_SUPPRESS_FIB_PENDING     (1 << 26)
-#define BGP_FLAG_SUPPRESS_DUPLICATES      (1 << 27)
+#define BGP_FLAG_SHUTDOWN (1 << 25)
+#define BGP_FLAG_SUPPRESS_FIB_PENDING (1 << 26)
+#define BGP_FLAG_SUPPRESS_DUPLICATES (1 << 27)
 #define BGP_FLAG_PEERTYPE_MULTIPATH_RELAX (1 << 29)
 
 	/* BGP default address-families.
 	 * New peers inherit enabled afi/safis from bgp instance.
 	 */
 	uint16_t default_af_flags;
-#define BGP_DEFAULT_AF_IPV4_UNICAST            (1 << 0)
-#define BGP_DEFAULT_AF_IPV4_MULTICAST          (1 << 1)
-#define BGP_DEFAULT_AF_IPV4_MPLS_VPN           (1 << 2)
-#define BGP_DEFAULT_AF_IPV4_FLOWSPEC           (1 << 3)
-#define BGP_DEFAULT_AF_IPV6_UNICAST            (1 << 4)
-#define BGP_DEFAULT_AF_IPV6_MULTICAST          (1 << 5)
-#define BGP_DEFAULT_AF_IPV6_MPLS_VPN           (1 << 6)
-#define BGP_DEFAULT_AF_IPV6_FLOWSPEC           (1 << 7)
-#define BGP_DEFAULT_AF_L2VPN_EVPN              (1 << 8)
+#define BGP_DEFAULT_AF_IPV4_UNICAST (1 << 0)
+#define BGP_DEFAULT_AF_IPV4_MULTICAST (1 << 1)
+#define BGP_DEFAULT_AF_IPV4_MPLS_VPN (1 << 2)
+#define BGP_DEFAULT_AF_IPV4_FLOWSPEC (1 << 3)
+#define BGP_DEFAULT_AF_IPV6_UNICAST (1 << 4)
+#define BGP_DEFAULT_AF_IPV6_MULTICAST (1 << 5)
+#define BGP_DEFAULT_AF_IPV6_MPLS_VPN (1 << 6)
+#define BGP_DEFAULT_AF_IPV6_FLOWSPEC (1 << 7)
+#define BGP_DEFAULT_AF_L2VPN_EVPN (1 << 8)
 
 	enum global_mode GLOBAL_GR_FSM[BGP_GLOBAL_GR_MODE]
 				      [BGP_GLOBAL_GR_EVENT_CMD];

```

</p>
</details>

            
If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

polychaeta reviewed (dismissed):
Thanks for your contribution to FRR!

* `sprintf, strcat, strcpy, inet_ntoa` are banned; please use `snprintf, strlcat, strlcpy, inet_ntop`

<details>
<summary><b>Click for style suggestions</b></summary>

To apply these suggestions:

<p>

```
curl -s https://gist.githubusercontent.com/polychaeta/535ca8f80d21b93b8418cdf665de9b0a/raw/19a1e86a4aa0c1ba66271f570f0d42aa59fe3480/cr_8401_1620050788.diff | git apply
```

</p>

<p>

```diff
diff --git a/bgpd/bgpd.h b/bgpd/bgpd.h
index 1e6624fc9..14e8141e1 100644
--- a/bgpd/bgpd.h
+++ b/bgpd/bgpd.h
@@ -450,50 +450,50 @@ struct bgp {
 #define BGP_FLAG_DETERMINISTIC_MED        (1 << 1)
 #define BGP_FLAG_MED_MISSING_AS_WORST     (1 << 2)
 #define BGP_FLAG_MED_CONFED               (1 << 3)
-#define BGP_FLAG_NO_CLIENT_TO_CLIENT      (1 << 4)
-#define BGP_FLAG_COMPARE_ROUTER_ID        (1 << 5)
-#define BGP_FLAG_ASPATH_IGNORE            (1 << 6)
-#define BGP_FLAG_IMPORT_CHECK             (1 << 7)
-#define BGP_FLAG_NO_FAST_EXT_FAILOVER     (1 << 8)
-#define BGP_FLAG_LOG_NEIGHBOR_CHANGES     (1 << 9)
+#define BGP_FLAG_NO_CLIENT_TO_CLIENT (1 << 4)
+#define BGP_FLAG_COMPARE_ROUTER_ID (1 << 5)
+#define BGP_FLAG_ASPATH_IGNORE (1 << 6)
+#define BGP_FLAG_IMPORT_CHECK (1 << 7)
+#define BGP_FLAG_NO_FAST_EXT_FAILOVER (1 << 8)
+#define BGP_FLAG_LOG_NEIGHBOR_CHANGES (1 << 9)
 
 /* This flag is set when we have full BGP Graceful-Restart mode enable */
-#define BGP_FLAG_GRACEFUL_RESTART         (1 << 10)
+#define BGP_FLAG_GRACEFUL_RESTART (1 << 10)
 
-#define BGP_FLAG_ASPATH_CONFED            (1 << 11)
-#define BGP_FLAG_ASPATH_MULTIPATH_RELAX   (1 << 12)
+#define BGP_FLAG_ASPATH_CONFED (1 << 11)
+#define BGP_FLAG_ASPATH_MULTIPATH_RELAX (1 << 12)
 #define BGP_FLAG_RR_ALLOW_OUTBOUND_POLICY (1 << 13)
 #define BGP_FLAG_DISABLE_NH_CONNECTED_CHK (1 << 14)
-#define BGP_FLAG_MULTIPATH_RELAX_AS_SET   (1 << 15)
-#define BGP_FLAG_FORCE_STATIC_PROCESS     (1 << 16)
-#define BGP_FLAG_SHOW_HOSTNAME            (1 << 17)
-#define BGP_FLAG_GR_PRESERVE_FWD          (1 << 18)
-#define BGP_FLAG_GRACEFUL_SHUTDOWN        (1 << 19)
-#define BGP_FLAG_DELETE_IN_PROGRESS       (1 << 20)
-#define BGP_FLAG_SELECT_DEFER_DISABLE     (1 << 21)
-#define BGP_FLAG_GR_DISABLE_EOR           (1 << 22)
-#define BGP_FLAG_EBGP_REQUIRES_POLICY     (1 << 23)
-#define BGP_FLAG_SHOW_NEXTHOP_HOSTNAME    (1 << 24)
+#define BGP_FLAG_MULTIPATH_RELAX_AS_SET (1 << 15)
+#define BGP_FLAG_FORCE_STATIC_PROCESS (1 << 16)
+#define BGP_FLAG_SHOW_HOSTNAME (1 << 17)
+#define BGP_FLAG_GR_PRESERVE_FWD (1 << 18)
+#define BGP_FLAG_GRACEFUL_SHUTDOWN (1 << 19)
+#define BGP_FLAG_DELETE_IN_PROGRESS (1 << 20)
+#define BGP_FLAG_SELECT_DEFER_DISABLE (1 << 21)
+#define BGP_FLAG_GR_DISABLE_EOR (1 << 22)
+#define BGP_FLAG_EBGP_REQUIRES_POLICY (1 << 23)
+#define BGP_FLAG_SHOW_NEXTHOP_HOSTNAME (1 << 24)
 
 /* This flag is set if the instance is in administrative shutdown */
-#define BGP_FLAG_SHUTDOWN                 (1 << 25)
-#define BGP_FLAG_SUPPRESS_FIB_PENDING     (1 << 26)
-#define BGP_FLAG_SUPPRESS_DUPLICATES      (1 << 27)
+#define BGP_FLAG_SHUTDOWN (1 << 25)
+#define BGP_FLAG_SUPPRESS_FIB_PENDING (1 << 26)
+#define BGP_FLAG_SUPPRESS_DUPLICATES (1 << 27)
 #define BGP_FLAG_PEERTYPE_MULTIPATH_RELAX (1 << 29)
 
 	/* BGP default address-families.
 	 * New peers inherit enabled afi/safis from bgp instance.
 	 */
 	uint16_t default_af_flags;
-#define BGP_DEFAULT_AF_IPV4_UNICAST            (1 << 0)
-#define BGP_DEFAULT_AF_IPV4_MULTICAST          (1 << 1)
-#define BGP_DEFAULT_AF_IPV4_MPLS_VPN           (1 << 2)
-#define BGP_DEFAULT_AF_IPV4_FLOWSPEC           (1 << 3)
-#define BGP_DEFAULT_AF_IPV6_UNICAST            (1 << 4)
-#define BGP_DEFAULT_AF_IPV6_MULTICAST          (1 << 5)
-#define BGP_DEFAULT_AF_IPV6_MPLS_VPN           (1 << 6)
-#define BGP_DEFAULT_AF_IPV6_FLOWSPEC           (1 << 7)
-#define BGP_DEFAULT_AF_L2VPN_EVPN              (1 << 8)
+#define BGP_DEFAULT_AF_IPV4_UNICAST (1 << 0)
+#define BGP_DEFAULT_AF_IPV4_MULTICAST (1 << 1)
+#define BGP_DEFAULT_AF_IPV4_MPLS_VPN (1 << 2)
+#define BGP_DEFAULT_AF_IPV4_FLOWSPEC (1 << 3)
+#define BGP_DEFAULT_AF_IPV6_UNICAST (1 << 4)
+#define BGP_DEFAULT_AF_IPV6_MULTICAST (1 << 5)
+#define BGP_DEFAULT_AF_IPV6_MPLS_VPN (1 << 6)
+#define BGP_DEFAULT_AF_IPV6_FLOWSPEC (1 << 7)
+#define BGP_DEFAULT_AF_L2VPN_EVPN (1 << 8)
 
 	enum global_mode GLOBAL_GR_FSM[BGP_GLOBAL_GR_MODE]
 				      [BGP_GLOBAL_GR_EVENT_CMD];

```

</p>
</details>

            
If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

taspelund commented:
@ton31337 @rwestphal I updated the test and moved all the `bgp default <afi-safi>` commands into a single DEFPY that uses a helper function.  Please let me know if there's more I can fix up :)

ton31337 commented on bgpd/bgp_vty.c:
```diff
@@ -3689,52 +3749,50 @@ DEFPY (no_bgp_bestpath_bw,
-	SET_FLAG(bgp->flags, BGP_FLAG_DEFAULT_IPV6);
-	return CMD_SUCCESS;
-}
+	strcpy(afi_safi_str, afi_safi);
```
strlcpy, please.

I would use `bgp default <afi> <safi>` instead of `bgp default <afi>-<safi>` at all. Or is here a real reason for that except that `bgp default ipv4-unicast` exists?

taspelund replied:
yep, just saw the strlcpy warning so I'm fixing that now.

yes, that's the exact reason I'm maintaining the hyphen.  I also think two separate args is cleaner, but ipv4-unicast has existed for a while and I didn't want to break backwards compatibility or introduce inconsistency in the UI.

taspelund replied:
Just did a force push that resolves this (and moves from strtok -> strtok_r).

ton31337 reviewed (commented):
A couple of questions :)

ton31337 commented on bgpd/bgp_vty.c:
```diff
@@ -16982,40 +17040,125 @@ static void bgp_config_write_peer_af(struct vty *vty, struct bgp *bgp,
-				if (afi == AFI_IP
-				    && CHECK_FLAG(bgp->flags,
-						  BGP_FLAG_NO_DEFAULT_IPV4)) {
+			if (afi == AFI_IP) {
```
Wouldn't be easier with `bool default_af[AFI_MAX][SAFI_MAX];` here instead of default_af_flags? I mean by iterating like:

```
if (!bgp->default_af[afi][safi])
	vty_out(vty, "  neighbor %s activate\n",
				addr);
```
etc.

ton31337 replied:
ping @taspelund :)

taspelund replied:
Sorry, got caught up with that pesky dayjob ;-)

For this my intention was mainly to try and match the existing logic + extend it for other AFIs.
I'm not opposed to using a 2D array, but my reason for not using it was simply because it didn't occur to me at the time.

If you think this is the proper way to get this done, lmk and I can rewrite it to use that instead.

ton31337 replied:
I still think this is a better way to handle that with much less code and less chance to hit the bug in the future :shrug:

ton31337 reviewed (changes requested):
Please rename tests/topotest directories to follow new guidelines:

```
Directory name for a new topotest must not contain hyphen (``-``) characters.
  To separate words, use underscores (``_``). For example, `tests/topotests/bgp_new_example`.
```

taspelund commented:
> Please rename tests/topotest directories to follow new guidelines:
> 
> ```
> Directory name for a new topotest must not contain hyphen (``-``) characters.
>   To separate words, use underscores (``_``). For example, `tests/topotests/bgp_new_example`.
> ```

Sure.  I'll fix this up and force push.

taspelund commented:
ci:rerun

ton31337 reviewed (commented):
LGTM, just reformat Python code using `black` and :+1:

taspelund commented:
Just force pushed after adding `black` formatting.  Thanks @ton31337 :-)

taspelund commented:
ci:rerun
